internal/utils/pagination/paginator.go

```diff
@@ -1,6 +1,6 @@
-package currencycloud
+package pagination
 
-func shouldFetchMore[T any](ret []T, nextPage, pageSize int) (bool, bool, []T) {
+func ShouldFetchMore[T any](ret []T, nextPage, pageSize int) (bool, bool, []T) {
```

Comment:
This function is very specific to currencycloud with the nextPage being at -1 when there are no more pages, which can not be the case in other connectors (and then create some errors). 

We can maybe have a more generic version of the function, and use a specific one for currencycloud ?

Reply:
Alright. I'll split the functions.